### PR TITLE
Add notes on configuring LDAP STS with Microsoft Active Directory

### PR DESCRIPTION
**Pull request by donatello**
## Motivation and Context

Tested with Microsoft AD.

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

File: docs/sts/ldap.md [hunk 1]
```diff
@@ -90,3 +90,50 @@ The **MINIO_IDENTITY_LDAP_GROUP_SEARCH_FILTER** and
 **MINIO_IDENTITY_LDAP_GROUP_SEARCH_BASE_DN** environment variables
 support substitution of the *username* and *usernamedn* variables
 only.
+
+## Notes on configuring with Microsoft Active Directory (AD)
+
+The LDAP STS API also works with Microsoft AD and can be configured in a manner
+similar to the above. The following are some notes on determining the values of
```
Comment (eco-minio) [inline docs/sts/ldap.md]:
```suggestion
as above. The following are some notes on determining the values of
```

File: docs/sts/ldap.md [hunk 2]
```diff
@@ -90,3 +90,50 @@ The **MINIO_IDENTITY_LDAP_GROUP_SEARCH_FILTER** and
 **MINIO_IDENTITY_LDAP_GROUP_SEARCH_BASE_DN** environment variables
 support substitution of the *username* and *usernamedn* variables
 only.
+
+## Notes on configuring with Microsoft Active Directory (AD)
+
+The LDAP STS API also works with Microsoft AD and can be configured in a manner
```
Comment (eco-minio) [inline docs/sts/ldap.md]:
```suggestion
The LDAP STS API also works with Microsoft AD and can be configured
```

**Review by eco-minio — CHANGES_REQUESTED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :x: [more...](#3f1e3fba-6fa6-41e7-829d-f04aff06da99) |
<a name="3f1e3fba-6fa6-41e7-829d-f04aff06da99"></a>
### 8260-49eaa4a/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:31048
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0c83daf8bb63:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 2 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 29 seconds
(8/14) Running minio-dotnet tests ... done in 25 seconds
(9/14) Running minio-go tests ... done in 55 seconds
(10/14) Running minio-java tests ... done in 14 seconds
(11/14) Running minio-js tests ... done in 39 seconds
(12/14) Running minio-py tests ... FAILED in 1 minutes and 2 seconds
{
  "duration": 5021,
  "status": "FAIL",
  "name": "minio-py:test_presigned_put_object_expiry_5sec",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1489, in test_presigned_put_object_expiry_5sec\n    raise ValueError('Presigned put url failed to expire!')\nValueError: Presigned put url failed to expire!\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2045, in main\n    test_presigned_put_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1491, in test_presigned_put_object_expiry_5sec\n    raise Exception(err)\nException: Presigned put url failed to expire!\n",
  "function": "presigned_put_object(bucket_name, object_name, expires)",
  "args": {
    "object_name": "668a784f-02b1-4a13-b28d-a9543def0862",
    "bucket_name": "minio-py-test-743214e6-1d56-44bd-97cf-c594f0e2eecf"
  },
  "message": "Presigned put url failed to expire!"
}

Executed 11 out of 14 tests successfully.
```

**Comment by harshavardhana** [comment]
ping @eco-minio

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#543f605b-3424-447b-a61c-e94b8b2464ee) |
| mint-compression-dist-xl.sh | :x: [more...](#e189f092-7f07-4525-ba36-bb922f9db9c7) |
| mint-fs.sh | :x: [more...](#c247fa0d-63c2-429c-a069-e3b3ea4f72bd) |
| mint-dist-xl.sh | :x: [more...](#9bc7d2a4-f32c-4efb-a920-46d84b991e08) |
<a name="543f605b-3424-447b-a61c-e94b8b2464ee"></a>
### 8260-1bdb7c5/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:31807
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b6c994c6934e:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-ntnh9jyryc3mkb",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364782,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15C559F1674F530A, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="e189f092-7f07-4525-ba36-bb922f9db9c7"></a>
### 8260-1bdb7c5/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:30868
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7caebb3d1e0f:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 2 minutes and 7 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 45 seconds
(8/14) Running minio-dotnet tests ... done in 35 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 20 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 34 seconds
(11/14) Running minio-js tests ... FAILED in 18 seconds
{
  "name": "minio-js",
  "function": "fPutObject(bucketName, objectName, filePath, metaData)",
  "args": "bucketName:minio-js-test-1b485f56-91a0-48b4-b0de-6a5e9a47a82f, objectName:datafile-65-MB, filePath:/tmp/datafile-65-MB",
  "duration": 1695,
  "status": "FAIL",
  "error": "Error: write EPIPE at exports._errnoException (util.js:1020:11) at WriteWrap.afterWrite (net.js:806:14)"
}

Executed 10 out of 14 tests successfully.
```
<a name="9bc7d2a4-f32c-4efb-a920-46d84b991e08"></a>
### 8260-1bdb7c5/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:30525
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d0fb2461e5c5:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 45 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 18 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 39 seconds
(8/14) Running minio-dotnet tests ... done in 44 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 47 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 41 seconds
(11/14) Running minio-js tests ... done in 56 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 27 seconds
{
  "args": {
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-09-17T22-03-24Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15C55A3C6B837E36\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Tue, 17 Sep 2019 22:27:40 GMT\"]}}",
    "response.reason": "Forbidden",
    "presigned_get_object_url": "http://72.28.97.51:30525/minio-py-test-f8fd8438-44b5-427b-a652-9c48c5a392f4/f623ab64-6143-40e6-a4ec-6d62a57c2109?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20190917%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20190917T222735Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=ccb53ffb9d7ed8ccc16dae39f955fa5b37450bdfd2346642f94f3c4a95fb619e",
    "response.status": 403,
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>f623ab64-6143-40e6-a4ec-6d62a57c2109</Key><BucketName>minio-py-test-f8fd8438-44b5-427b-a652-9c48c5a392f4</BucketName><Resource>/minio-py-test-f8fd8438-44b5-427b-a652-9c48c5a392f4/f623ab64-6143-40e6-a4ec-6d62a57c2109</Resource><RequestId>15C55A3C6B837E36</RequestId><HostId>df3dfc99-e4c5-422e-9fec-34b96655e5c2</HostId></Error>",
    "bucket_name": "minio-py-test-f8fd8438-44b5-427b-a652-9c48c5a392f4",
    "object_name": "f623ab64-6143-40e6-a4ec-6d62a57c2109"
  },
  "duration": 89,
  "status": "FAIL",
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1341, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "message": "AccessDenied: message: Access Denied"
}

Executed 11 out of 14 tests successfully.
```
<a name="c247fa0d-63c2-429c-a069-e3b3ea4f72bd"></a>
### 8260-1bdb7c5/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:32522
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7b514bce90d9:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 1 seconds
(5/14) Running awscli tests ... done in 1 minutes and 51 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 8 seconds
{
  "name": "mc",
  "duration": "4817",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-29610 --quiet --no-color cp myminio/mc-test-bucket-31924/mc-test-object-10107 mc-test-object-10107.downloaded  >>> \n`myminio/mc-test-bucket-31924/mc-test-object-10107` -> `mc-test-object-10107.downloaded`\nmc: <ERROR> Failed to copy `http://72.28.97.59:32522/mc-test-bucket-31924/mc-test-object-10107`. read tcp 172.17.0.2:49208->72.28.97.59:32522: read: connection reset by peer\nmc: <ERROR> Session safely terminated. To resume session `mc session resume hyOfbLtI`"
}

Executed 6 out of 14 tests successfully.
```

**Review by eco-minio — APPROVED** [review]
LGTM

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#84c95f09-ad34-4bb3-9571-312481a913ce) |
<a name="84c95f09-ad34-4bb3-9571-312481a913ce"></a>
### 8260-80b89fe/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:31964
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp dc9d19a56f06:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 55 seconds
(4/14) Running aws-sdk-ruby tests ... done in 18 seconds
(5/14) Running awscli tests ... done in 2 minutes and 41 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 1 minutes and 11 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 31 seconds
(9/14) Running minio-go tests ... FAILED in 8 minutes and 41 seconds
{
  "args": {},
  "duration": 81661,
  "error": "The specified key does not exist.",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "Stat failed",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#554869de-a839-404c-b872-629ee5e23fe9) |
| mint-large-bucket.sh | :x: [more...](#36cfaf67-d270-4844-8157-dd0361093226) |
| mint-compression-dist-xl.sh | :x: [more...](#66d5b8ef-55f7-4ae7-a5f6-81dbe2531b30) |
<a name="66d5b8ef-55f7-4ae7-a5f6-81dbe2531b30"></a>
### 8260-3f398c8/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:32544
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 08f159af2a3b:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 5 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 45 seconds
(8/14) Running minio-dotnet tests ... done in 33 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 57 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 13 seconds
(11/14) Running minio-js tests ... FAILED in 23 seconds
{
  "name": "minio-js",
  "function": "fGetObject(bucketName, objectName, filePath, callback)",
  "args": "bucketName:minio-js-test-7f6799db-7874-4a96-8ae3-88961eb517ba, objectName:datafile-65-MB, filePath:/tmp/datafile-65-MB.download",
  "duration": 5446,
  "status": "FAIL",
  "error": "Error: Size mismatch between downloaded file and the object at /mint/run/core/minio-js/node_modules/minio/dist/main/minio.js:965:12 at nextTask (node_modules/minio/node_modules/async/dist/async.js:4578:13) at next (node_modules/minio/node_modules/async/dist/async.js:4586:13) at /mint/run/core/minio-js/node_modules/minio/node_modules/async/dist/async.js:325:20 at FSReqWrap.oncomplete (fs.js:123:15)"
}

Executed 10 out of 14 tests successfully.
```
<a name="36cfaf67-d270-4844-8157-dd0361093226"></a>
### 8260-3f398c8/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:31142
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp fb6741b53e12:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 11 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 59 seconds
(4/14) Running aws-sdk-ruby tests ... done in 20 seconds
(5/14) Running awscli tests ... done in 2 minutes and 16 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 1 minutes and 2 seconds
(8/14) Running minio-dotnet tests ... FAILED in 3 minutes and 5 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (The operation has timed out.) ---> System.Net.WebException: The operation has timed out.
   at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at Minio.Functional.Tests.FunctionalTest.PresignedGetObject_Test1(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 2086
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.Program.Main(String[] args) in /mint/run/core/minio-dotnet/Program.cs:line 129

Executed 7 out of 14 tests successfully.
```
<a name="554869de-a839-404c-b872-629ee5e23fe9"></a>
### 8260-3f398c8/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30074
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b31a9a9666c8:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-p23muxfuagp61h",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365151,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15C561AEF7DCCEDB, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Pull request merged**